src/lib/sedlexing.ml

```diff
@@ -70,17 +114,24 @@ let empty_lexbuf =
 let dummy_uchar = Uchar.of_int 0
 let nl_uchar = Uchar.of_int 10
 
-let create refill =
-  { empty_lexbuf with refill; buf = Array.make chunk_size dummy_uchar }
+let create ?(bytes_per_char = fun _ -> 1) refill =
+  {
+    (empty_lexbuf bytes_per_char) with
+    refill;
+    buf = Array.make chunk_size dummy_uchar;
+  }
 
-let set_position lexbuf position =
+let set_position ?bytes_position lexbuf position =
   lexbuf.offset <- position.Lexing.pos_cnum - lexbuf.pos;
   lexbuf.curr_bol <- position.Lexing.pos_bol;
-  lexbuf.curr_line <- position.Lexing.pos_lnum
+  lexbuf.curr_line <- position.Lexing.pos_lnum;
+  let bytes_position = Option.value ~default:position bytes_position in
+  lexbuf.bytes_offset <- bytes_position.Lexing.pos_cnum - lexbuf.pos;
```

Comment:
should this be  `lexbuf.bytes_offset <- bytes_position.Lexing.pos_cnum - lexbuf.bytes_pos;` ?